### PR TITLE
JENKINS-62285 Don't use browser alert for notifications and confirmation

### PR DESCRIPTION
**Pull request by aytuncbeken**
Hi,
As a part of the UI Hackfest. I fixed the issue JENKINS-62285.
I hope this PR is acceptable for you.

You can see the details of the issue from here.
https://issues.jenkins-ci.org/browse/JENKINS-62285

### CODE REVIEW

File: src/main/webapp/js/addrole.js [hunk 1]
```diff
@@ -100,10 +100,10 @@ const sendPostRequest = (postUrl, json) => {
 
     xhr.onload = () => {
         if (xhr.status === 200) {
-            alert('The role was added successfully');
-            location.reload(); // refresh the page
+            showNotificationOK('The role was added successfully');
+            //location.reload(); // refresh the page
```
Comment (timja) [inline src/main/webapp/js/addrole.js:104]: we shouldn't leave commented out code in, it looks like the function handles this?

```suggestion
```

**Comment by timja** [comment]
It would be best to name your PR with a summary description of what this PR does

File: src/main/webapp/js/addrole.js (hunk 1 above)
Comment (aytuncbeken) [reply src/main/webapp/js/addrole.js]: I forgot that, removed now

File: src/main/webapp/js/managesids.js [hunk 2]
```diff
@@ -59,15 +58,14 @@ function removeSid(roleType, roleName, sidInputBoxId) {
     request.open('POST', url);
     request.onload = () => {
         if (request.status === 200) {
-            alert('Sid removed successfully.');
-            location.reload();
-        } else {
-            alert('Unable to remove the sid.' + request.responseText);
+            showNotificationOK('Sid removed successfully.');
+        } else {s
```
Comment (timja) [inline src/main/webapp/js/managesids.js:62]: typo
```suggestion
        } else {
```

Comment (aytuncbeken) [reply src/main/webapp/js/managesids.js]: fixed

File: src/main/webapp/js/notification.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    notificationBar.show(message, options);
+    if (reload) {
+        window.setTimeout(function () {
+            location.reload();
+        }, notificationBar.DELAY);
+    }
+}
+
+function showNotificationOK(message) {
+    showNotification(message, notificationBar.OK, true);
```
Comment (AbhyudayaSharma) [inline src/main/webapp/js/notification.js:11]: There's a small problem with this behaviour. The page only refreshes when the success notification disappears. So for about 2 seconds, the notification shows but the content is not updated. With the alerts, as soon as the user clicked OK, the page was refreshed. Would it be possible to use a modal popup here? Also note that I'm working on #58 which will completely change the UI.

Comment (aytuncbeken) [reply src/main/webapp/js/notification.js:11]: Hi, thanks for your comment. I will search for modal solution, not sure though if its possible or not. For providing similar functionality I added optional reload feature, which waits for notification.DELAY.

About the UI change, of course It is up to you. We can apply this change until the new UI is published.

@timja Do you know any examples for Modal implementation ? Thanks.

Comment (timja) [reply src/main/webapp/js/notification.js:11]: better to not refresh the page and to do a live update, or let the user refresh it themselves really.

Comment (aytuncbeken) [reply src/main/webapp/js/notification.js:11]: I think that is topic of the UI change that @AbhyudayaSharma mentioned. Should we decline this request then ?

Comment (timja) [reply src/main/webapp/js/notification.js:11]: Yeah, the code will likely still be useful, thanks for doing this!

**Comment by AbhyudayaSharma** [comment]
Thanks @aytuncbeken ! This code will be useful for me when working on #58. I'm closing this PR then.

File: src/main/webapp/js/notification.js (hunk 3 above)
Comment (aytuncbeken) [reply src/main/webapp/js/notification.js:11]: No problem

**Pull request closed without merging**